### ISSUE TITLE
[Supplier] Provide config/spec file when staking Supplier

### ISSUE THREAD

**Issue by Olshansk** [pokt-network/poktroll#137]
## Objective

Productionize the interface via which supplier will be staking for numerous complex services

## Origin Document

- Should be somewhat similar to how we enabled specifying requests via a flag in the [v1 repo](https://github.com/pokt-network/pocket)
Related to #138
- The following snippet from [https://github.com/pokt-network/poktroll/blob/main/x/supplier/client/cli/tx_stake_supplier.go](x/supplier/client/cli/tx_stake_supplier.go) captures it:

```go
func CmdStakeSupplier() *cobra.Command {
	// fromAddress & signature is retrieved via `flags.FlagFrom` in the `clientCtx`
	cmd := &cobra.Command{
		// TODO_HACK: For now we are only specifying the service IDs as a list of of strings separated by commas.
		// This needs to be expand to specify the full SupplierServiceConfig. Furthermore, providing a flag to
		// a file where SupplierServiceConfig specifying full service configurations in the CLI by providing a flag that accepts a JSON string
		Use:   "stake-supplier [amount] [svcId1;url1,svcId2;url2,...,svcIdN;urlN]",
		Short: "Stake a supplier",
		Long: `Stake an supplier with the provided parameters. This is a broadcast operation that
will stake the tokens and associate them with the supplier specified by the 'from' address.
TODO_HACK: Until proper service configuration files are supported, suppliers must specify the services as a single string
of comma-separated values of the form 'service;url' where 'service' is the service ID and 'url' is the service URL.
For example, an application that stakes for 'anvil' could be matched with a supplier staking for 'anvil;http://anvil:8547'.
```

## Goals

- Set the foundation for how complex specification & configuration files for supplier staking will be provided
- Enable node runners to stake for repeatedly staking for complex services

## Deliverables

- [ ] A PR that modifies how `supplier stake` functions by providing a path to a file on disk
- [ ] An ability to provide the complete complex configuration via an inline json file
- [ ] Cleanup of all downstream code (e.g. message preparation) when staking for the service
- [ ] Unit tests & validation for all existing fields & parametesr

## Non-goals / Non-deliverables

- Adding/removing configurations to the `SupplierServiceConfig`

## General deliverables

- [ ] **Comments**: Add/update TODOs and comments alongside the source code so it is easier to follow.
- [ ] **Testing**: Add new tests (unit and/or E2E) to the test suite.
- [ ] **Makefile**: Add new targets to the Makefile to make the new functionality easier to use.
- [ ] **Documentation**: Update architectural or development READMEs; use [mermaid](https://mermaid-js.github.io/mermaid/) diagrams where appropriate.

---

**Creator**: @Olshansk 
**Co-Owners**: @h5law

**Comment by Olshansk** [comment]
In #177, we introduced this flag: `cmd.Flags().StringVar(&flagSmtStorePath, "smt-store", "smt", "Path to the SMT KV store")`. This is an example of something that should be part of the supplier's config file so we don't end up with dozens of flags.

[For reference, in v0](https://docs.pokt.network/node/tutorials/zero-to-node/pocket-config/), we introduced both `config.json` and `chains.json` files and belive that something similar would be appropriate in Shannon as well.

![Screenshot 2023-11-13 at 5 07 40 PM](https://github.com/pokt-network/poktroll/assets/1892194/b8e826fc-32e7-4e93-8cd5-f75023ca15c1)

cc @red-0ne @h5law for context
